cascade-merge/action.yml

```diff
@@ -0,0 +1,198 @@
+    required: false
+  create-pr:
+    description: 'create PR '
+    required: true
+    default: 'true'
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0  # fetch all history so we can push to other branches
+
+    - name: Skip cascade merge
+      run: |
+        function echo_green {
+            echo -e "\e[32m$1\e[0m"
+        }
+        echo_green "The job was skipped because the commit message contains '[skip cascade merge]'"
+      if: contains(env.COMMIT_MSG, '[skip cascade merge]')
+      shell: bash
+      env:
+        COMMIT_MSG: ${{ github.event.head_commit.message }}
+
+    - name: Configure Git
+      run: |
+        git config user.name "GitHub Actions"
+        git config user.email "github-actions@github.com"
+      shell: bash
+
+    - name: Detect conflicts
+      run: |
+        CASCADE_BRANCHES=( $(git for-each-ref --format="%(refname)" refs/remotes/origin/release | sed 's%refs/remotes/origin/%%g') )
+        CASCADE_BRANCHES[${#CASCADE_BRANCHES[@]}]="main"
+
+        # create the first branch
+        git checkout ${CASCADE_BRANCHES[0]}
+        for step in $(seq 0 $((${#CASCADE_BRANCHES[@]} - 2)))
+        do
+            src=${CASCADE_BRANCHES[${step}]}
+            dest=${CASCADE_BRANCHES[$((${step}+1))]}
+            git checkout $dest
+            msg="$dest <= $src"
+            git merge -m "$msg" $src || (echo " === $msg KO ==="; exit 1)
+            echo " === $msg OK ==="
+        done
+      shell: bash
+
+#    - name: Notify failing merge
+#      if: failure()
+#      uses: LedgerHQ/actions/slack-notification@main
+#      env:
+#        SLACK_WEBHOOK_URL: ${{ inputs.slack-webhook-url }}
+#        SLACK_API_TOKEN: ${{ inputs.slack-api-token }}
+#        SLACK_CUSTOM_PAYLOAD: |-
+#          {
+#            "blocks": [
+#              {
+#                "type": "header",
+#                "text": {
+#                  "type": "plain_text",
+#                  "text": ":boom: Merge conflict :boom:",
+#                  "emoji": true
+#                }
+#              },
+#              {
+#                "type": "section",
+#                "text": {
+#                  "type": "mrkdwn",
+#                  "text": "<@SLACK_MEMBER_ID> please fix it :iamwatchingyou:"
+#                }
+#              },
+#              {
+#                "type": "actions",
+#                "elements": [
+#                  {
+#                    "type": "button",
+#                    "text": {
+#                      "type": "plain_text",
+#                      "text": ":github: Github build",
+#                      "emoji": true
+#                    },
+#                    "url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+#                  }
+#                ]
+#              },
+#              {
+#                "type": "divider"
+#              }
+#            ]
+#          }
+#
```

Comment:
let's drop the commented stuff

Reply:
here am waiting for slack secrets to be shared as organization secrets https://ledgerhq.atlassian.net/servicedesk/customer/portal/14/ACCESS-3920?created=true

Reply:
got it